test/constructRoutes.test.js

```diff
@@ -0,0 +1,288 @@
+import { constructRoutes } from "../src/single-spa-layout.js";
+
+describe("constructRoutes", () => {
+  let registerApplication = jest.fn();
+
+  beforeEach(() => {
+    registerApplication.mockReset();
+  });
+
+  describe(`input validation`, () => {
+    it("accepts a valid routesConfig", () => {
+      constructRoutes({
+        mode: "history",
+        base: "/",
+        containerEl: "#selector",
+        routes: [
+          { type: "application", name: "@org/navbar" },
+          {
+            type: "route",
+            path: "app1",
+            routes: [
+              { type: "application", name: "@org/main-sidenav" },
+              { type: "application", name: "@org/app1" },
+            ],
+          },
+          {
+            type: "route",
+            path: "app2",
+            routes: [
+              { type: "application", name: "@org/main-sidenav" },
+              { type: "application", name: "@org/app2" },
+            ],
+          },
+          {
+            type: "route",
+            path: "settings",
+            routes: [{ type: "application", name: "@org/settings" }],
+          },
+          { type: "application", name: "@org/footer" },
+        ],
+      });
+    });
+  });
+
+  it(`throws an error if the first argument is not an object`, () => {
+    expect(() => {
+      constructRoutes();
+    }).toThrowError(/expected a plain object/);
+
+    expect(() => {
+      constructRoutes(null);
+    }).toThrowError(/expected a plain object/);
+
+    expect(() => {
+      constructRoutes("");
+    }).toThrowError(/expected a plain object/);
+
+    expect(() => {
+      constructRoutes(undefined);
+    }).toThrowError(/expected a plain object/);
+
+    expect(() => {
+      constructRoutes([]);
+    }).toThrowError(/expected a plain object/);
+  });
+
+  it(`throws an error if extra properties are provided`, () => {
+    expect(() => {
+      constructRoutes({
+        base: "/",
+        routes: [],
+        irrelevantProperty: "thing",
+      });
+    }).toThrowError("invalid properties");
```

Comment:
Probably something to consider in another PR, but I think it would be useful to call out which property is invalid here.

Reply:
Whoops. That's what I get for reading the tests before the code. 

You already have this behavior. 🙂